src/tactic/lift.lean

```diff
@@ -85,17 +85,16 @@ If the proof was not specified, we create assert it as a local constant.
 (The name of this local constant doesn't matter, since `lift` will remove it from the context.)
 -/
 meta def get_lift_prf (h : option pexpr) (old_tp new_tp inst e : expr)
-  (s : simp_lemmas) (to_unfold : list name) : tactic expr :=
-if h_some : h.is_some then
-  (do prf ← i_to_expr (option.get h_some), prf_ty ← infer_type prf,
+  (s : simp_lemmas) (to_unfold : list name) : tactic expr := do
   expected_prf_ty ← mk_app `can_lift.cond [old_tp, new_tp, inst, e],
-  unify prf_ty expected_prf_ty <|>
-    (do expected_prf_ty2 ← s.dsimplify to_unfold expected_prf_ty,
-      pformat!"lift tactic failed. The type of\n  {prf}\nis\n  {prf_ty}\nbut it is expected to be\n  {expected_prf_ty2}" >>= fail),
-  return prf)
-  else (do prf_nm ← get_unused_name,
-    prf ← mk_app `can_lift.cond [old_tp, new_tp, inst, e] >>= assert prf_nm,
-    dsimp_target s to_unfold {}, swap, return prf)
+  expected_prf_ty ← s.dsimplify to_unfold expected_prf_ty,
+  if h_some : h.is_some then
```

Comment:
A `match h with | some h => ... | none => ...` seems infinitely clearer here than the dependent if, but I realize that's in the eye of the beholder.

Reply:
Fair